pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -260,6 +266,15 @@ func (c *Cluster) makeDeployment(nodeName string, selection rookalpha.Selection,
 	if c.HostNetwork {
 		DNSPolicy = v1.DNSClusterFirstWithHostNet
 	}
+
+	lifecycle := &v1.Lifecycle{
```

Comment:
Since this is the same lifecycle for all the daemons, how about a common method to get it?

Reply:
> Could there be any issue with upgrades when changing from the root to ceph user?

The `chown` should prevent any issues, but let me do more testing tomorrow on this front. Also, we use the flag `--setuser-match-path` which will run under the same user as the ownership (if not `ceph`), so it's safe.